rust/differential_datalog/profile.rs

```diff
@@ -0,0 +1,109 @@
+//! Memory profile of a DDlog program.
```

Comment:
Isn't there some overhead for adding all this code, even if profiling is disabled?

Reply:
Should be negligible. The runtime callback is invoked once for every batch and basically just updates two keys in two different maps.